l10n_ar_sale/models/sale_order.py

```diff
@@ -63,13 +63,20 @@ def _get_sale_checkbook(self):
     @api.model_create_multi
     def create(self, vals):
         for val in vals:
+            sale_checkbook_id = val.get('sale_checkbook_id')
+            sale_checkbook_type = val.get('type_id')
             if self.env.user.has_group('l10n_ar_sale.use_sale_checkbook') and \
                 val.get('name', _('New')) == _('New') and \
-                    val.get('sale_checkbook_id'):
-                sale_checkbook = self.env['sale.checkbook'].browse(
-                    val.get('sale_checkbook_id'))
+                sale_checkbook_id:
+                sale_checkbook = self.env['sale.checkbook'].browse(sale_checkbook_id)
                 val['name'] = sale_checkbook.sequence_id and\
                     sale_checkbook.sequence_id._next() or _('New')
+            elif sale_checkbook_type and not sale_checkbook_id:
+                sale_checkbook_type = self.env['sale.order.type'].browse(val.get('type_id')).sale_checkbook_id
+                # Obtenemos el talonario asociado al tipo de venta, esto lo utilizamos si la venta viene de website
+                # O el usuario no tiene acceso a los talonarios de venta
+                val['name'] = sale_checkbook_type and sale_checkbook_type.sequence_id and\
+                    sale_checkbook_type.sequence_id._next() or _('New')
```

Comment:
```suggestion
            elif sale_checkbook_type and not sale_checkbook_id:
                sale_checkbook = self.env['sale.order.type'].browse(val.get('type_id')).sale_checkbook_id
                # Obtenemos el talonario asociado al tipo de venta, esto lo utilizamos si la venta viene de website
                # O el usuario no tiene acceso a los talonarios de venta
                val['name'] = sale_checkbook.sequence_id and\
                    sale_checkbook_type.sequence_id._next() or _('New')
```
por que no agregaríamos not self.env.user.has_group('l10n_ar_sale.use_sale_checkbook') en el elif?

Reply:
en la línea 75 existe la posibilidad de que "self.env['sale.order.type'].browse(val.get('type_id'))" de False? en ese caso si haces False.sale_checkbook_id va a arrojar piedrazo

Reply:
también sugiero como mostré en el diff sugerido cambiar el nombre de la variable sale_checkbook_type a sale_checkbook ya que estás guardando en la variable el sale_checkbook_id y no el sale_checkbook_type